### ISSUE TITLE
Video speed problem

### ISSUE THREAD

**Issue by KittyNeverDies** [libre-tube/LibreTube#5256]
### Steps to reproduce

Can't be reproduce

### Expected behavior

Play video with stability internet connection

### Actual behavior

- All piped instances working very slow and cutted and stoping every one-two second of video (on 144p or play only audio don't fix the issue)
- Play video directly from YouTube don't work (prob. because I am in Russia)

### LibreTube version

Latest

### Android version

Android 13

### Other details

It is in Russia country

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Tunsinasro** [comment]
I'm experiencing the same behavior

**Comment by Bnyro** [comment]
https://github.com/TeamPiped/Piped/issues/3196

**Comment by Mahfuj555** [comment]
Same problem i'm facing from Bangladesh 
Android version 6

**Comment by vagosss** [comment]
Same issue for the past two days.

**Comment by IndusAryan** [comment]
same problem, video buffer occurs and speed can be shown in traffic indicator and after that player pauses at 1 second and black screen. even with proxy off

**Comment by MyWay** [comment]
Same here

**Comment by sebastien46** [comment]
Please stop necro'ing, the issue's being resolved upstream.

**Comment by IndusAryan** [comment]
> Although my problem was solved by enabling HLS and LBRY HLS. Still, before I learned the trick though [this](https://github.com/TeamPiped/Piped/issues/3196), I stumbled upon an error log. Here are the details--
> ### Steps to Reproduce
> 
>     1. Enable Piped Proxy. (I was using privacydev.net)
> 
>     2. Disable Use HLS.
> 
>     3. Disable LBRY HLS.
> 
>     4. Play any video.
> 
> 
> ### Expected Behavior
> 
> Video should play normally.
> ### Actual Behavior
> 
> App crashes. Error log is shown.
> ### LibreTube Version
> 
> 0.20.1
> ### Android Version
> 
> Android 11
> ### Error Log
> 
> ```
> kotlinx.serialization.MissingFieldException: Fields [license, visibility, tags, metaInfo] are required for type with serial name 'com.github.libretube.api.obj.Streams', but they were missing at path: $
> 	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue$1(SourceFile:158)
> 	at kotlinx.serialization.json.Json.decodeFromString(SourceFile:26)
> 	at com.jakewharton.retrofit2.converter.kotlinx.serialization.DeserializationStrategyConverter.convert(SourceFile:33)
> 	at retrofit2.OkHttpCall.parseResponse(SourceFile:57)
> 	at retrofit2.OkHttpCall$1.onResponse(SourceFile:5)
> 	at androidx.appcompat.app.AlertController$2.onSuccess(SourceFile:34)
> 	at androidx.work.Worker$2.run(SourceFile:131)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
> 	at java.lang.Thread.run(Thread.java:923)
> 	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@b484091, Dispatchers.IO]
> Caused by: kotlinx.serialization.MissingFieldException: Fields [license, visibility, tags, metaInfo] are required for type with serial name 'com.github.libretube.api.obj.Streams', but they were missing
> 	at okio.-SegmentedByteString.throwMissingFieldException(SourceFile:115)
> 	at com.github.libretube.api.obj.Streams.<init>(SourceFile:1)
> 	at com.github.libretube.api.obj.Streams$$serializer.deserialize(SourceFile:1)
> 	at com.github.libretube.api.obj.Streams$$serializer.deserialize(SourceFile:2)
> 	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue$1(SourceFile:102)
> 	... 9 more
> ```

your problem has nothing to do with this, your instance is not updated, use official one

**Comment by luckkmaxx** [comment]
Same here. Tried everything. Sometimes enable HLS and goes well in home, then with mobile data fails again.
Piped web ui is working.
This happens with own and official instances.

**Comment by Enemyprovider** [comment]
Same here. It doesn't work

**Comment by akano12** [comment]
Not to sound passive aggressive but commenting "same/i have same issue" isn't going to fix the problem.

If you're facing same issue, kindly track the original issue at piped, you can find the link to the issue attached by Bnyro in first comment.

**Comment by akano12** [comment]
> Don't know if this would be considered as necroing, but the referenced issue does not really help in my case. HLS enabled or disabled, it doesn't work, LBRY HLS does nothing either. Even disabled piped proxy, used a VPN, used Orbot, nothing. All efforts are met with nothing but a "source error". Oddly enough the instances themselves work on a browser even without HLS.

The "solution" you mentioned is just a temporary workaround that wouldn't necessary work for everyone, I'm saying to follow the issue on piped because when there's a fix, it'll be pushed there.
There's nothing can be don't on client side (LibreTube)

**Comment by luckkmaxx** [comment]
Btw, since background play only works well in audio mode, and this bug makes to use HLS, audio mode and background playing is impossible to do.

**Comment by dreammirror000** [comment]
Same here on OnePlus 11 with public piped **and** personal piped. Cannot play anything.

**Comment by kzhe2059** [comment]
![ResizedImage_2023-12-04_07-27-56_6376](https://github.com/libre-tube/LibreTube/assets/29979557/21f4a3b9-fc4b-4ae6-835f-0683500d9c0f)
These two need to be enabled.

**Comment by MrPantherKing** [comment]
Thanks @Bnyro enabling HLS worked

**Comment by MrPusemukel** [comment]
Cool,  Enable HlS works for me, too

**Comment by DanielProg39** [comment]
Videos just started playing normally for me. BTW, I have HLS disabled

**Comment by MrPantherKing** [comment]
@Bnyro now i am receiving another message "initial video response invalid" and i am just getting a skeleton view of the player but video isnt playing again eve after enabling HLS

**Comment by FireMasterK** [comment]
> @Bnyro now i am receiving another message "initial video response invalid" and i am just getting a skeleton view of the player but video isnt playing again eve after enabling HLS

https://github.com/TeamPiped/Piped/issues/2487

TLDR: Your instance's IP is 'banned' by Google. Change it, or use a VPN/proxy.
